web-api/src/persistence/postgres/cases/getCaseByDocketNumber.ts

```diff
@@ -1,19 +1,104 @@
-import { getDbReader } from '@web-api/database';
-
-// Named getCaseByDocketNumberPostgres until the Dynamo getCaseByDocketNumber is removed
-export const getCaseByDocketNumberPostgres = async (docketNumber: string) => {
-  return await getDbReader(reader => {
-    return reader
-      .selectFrom('dwCase as c')
-      .leftJoin('dwWorkItem as d', 'd.docketNumber', 'c.docketNumber')
-      .leftJoin(
-        'dwCaseCorrespondence as co',
-        'c.docketNumber',
-        'co.docketNumber',
-      )
-      .where('c.docketNumber', '=', docketNumber)
-      .selectAll()
-      .select('c.docketNumber')
-      .execute();
+import { ConsolidatedCaseSummary } from '@shared/business/dto/cases/ConsolidatedCaseSummary';
+import { NotFoundError } from '@web-api/errors/errors';
+import { ServerApplicationContext } from '@web-api/applicationContext';
+import { aggregateCaseItems } from '@web-api/persistence/dynamo/helpers/aggregateCaseItems';
+import { getCasesMetadataWithCounselByLeadDocketNumber } from '@web-api/persistence/postgres/cases/getCasesMetadataWithCounselByLeadDocketNumber';
+import { getWorkItemsByDocketNumber } from '@web-api/persistence/postgres/workitems/getWorkItemsByDocketNumber';
+import { purgeDynamoKeys } from '@web-api/persistence/dynamo/helpers/purgeDynamoKeys';
+import { queryFull } from '@web-api/persistence/dynamodbClientService';
+import { UnknownAuthUser } from '@shared/business/entities/authUser/AuthUser';
+import { formatSealedAddresses } from '@shared/business/utilities/caseFilter';
+import { getCaseMetadataWithCounsel } from '@web-api/persistence/postgres/cases/getCaseMetadataWithCounsel';
+import { getCaseCorrespondenceByDocketNumber } from '@web-api/persistence/postgres/caseCorrespondences/getCaseCorrespondenceByDocketNumber';
+import { getCaseStatistics } from '@web-api/persistence/postgres/cases/statistics/getCaseStatistics';
+import { getCaseStatusHistory } from '@web-api/persistence/postgres/cases/getCaseStatusHistory';
+
+export const getCaseByDocketNumber = async ({
+  applicationContext,
+  docketNumber,
+  includeConsolidatedCases = true,
+  user = undefined, // Only needed to check permissions on sealed addresses for consolidated cases
+}: {
+  docketNumber: string;
+  applicationContext: ServerApplicationContext;
+  includeConsolidatedCases?: boolean;
+  user?: UnknownAuthUser;
+}): Promise<RawCase> => {
```

Comment:
In other Postgres work, we were able to return the Entity itself (e.g., Message) to avoid using the "Raw" entities. For Case, this is much more difficult because we choose different subsets of the Case table data in different situations. (For instance, sometimes we get practitioners, sometimes we don't. Sometimes we get the consolidated cases, sometimes we don't.) There are a few options to handle these discrepancies:

1. We create new entities for all the different subsets of data we need. This would involve a lot of work, and it is outside the scope of this already giant ticket.
2. We always get all Case data. This is inefficient.
3. We say, "Well, darn it," and use RawCase.

We opted for 3.